discord/utils.py

```diff
@@ -609,7 +609,7 @@ def _get_as_snowflake(data: Any, key: str) -> Optional[int]:
         return value and int(value)
 
 
-def _get_mime_type_for_image(data: bytes):
+def _get_mime_type_for_file(data: bytes):
```

Comment:
Please make this a new helper instead.

This change makes it so people can pass e.g. audio files for avatars and have it silently work when it would have failed prior.

Reply:
Okay, but then it needs two `_bytes_to_base64_...` functions to differ between image and audio files or a parameter for `_bytes_to_base64_data` to pass the type of data, e.g. `'image'` or `'audio'`. Otherwise I would have to call both functions in `_bytes_to_base64_data` and use try except to somehow get the mime type which doesn't solve the problem, e.g.:

```py
def _bytes_to_base64_data(data: bytes) -> str:
    # ...
    try:
        mime = _get_mime_type_for_image(data)
    except ValueError:
        try:
            mime = _get_mime_type_for_audio(data)
        except ValueError:
            raise ValueError('Unsupported file type given')
    # ...
```
Still allows passing audio files even image files are expected.

Reply:
I don't think it needs all that complication. You have `_get_mime_type_for_image` and `_get_mime_type_for_audio` then you inside `_bytes_to_base64_data` you can just do `audio=True` or `audio=False` (default) and appropriately branch it. I don't expect an endpoint to allow both forms of media so the try except is just noisy.

Reply:
Yeah exactly, I meant something like this. Of course the try except wouldn't solve the problem as I mentioned. 👍